### ISSUE TITLE
Support `targets` containing probabilities for each class

### ISSUE THREAD

**Issue by minhnguyent546** [itakurah/Focal-loss-PyTorch#2]
Is it possible to support `targets` containing probabilities for each class instead of class indicies [like this](https://docs.pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html#torch.nn.CrossEntropyLoss#:~:text=Probabilities,described%20as). In my case, I am using mixup&cutmix and targets will have shape (batch_size, num_classes) instead of (batch_size,) and currently not be able to incorporate mixup&cutmix with focal loss.

**Comment by itakurah** [comment]
Thanks for sharing the details. I’ll work on implementing a custom Focal Loss that supports soft targets like `CrossEntropyLoss` does, so it can be used seamlessly with MixUp and CutMix. I’ll update the code accordingly once it’s ready.

**Comment by minhnguyent546** [comment]
Thank you!

**Comment by itakurah** [comment]
@minhnguyent546 please check out the `feature_soft_labels` branch and let me know your feedback.

**Comment by minhnguyent546** [comment]
Thanks for the update. I have tested training my model using both Mixup & CutMix and Focal Loss, and it seems to be working correctly. However, there is a small issue validating `num_classes`, could you [take a look](https://github.com/minhnguyent546/Focal-loss-PyTorch/commit/ce37b21066cc1941c8c57fa59966a159b25029e2)?

**Comment by itakurah** [comment]
@minhnguyent546 thank you, I missed that! Could you please create a draft PR with the changes?

**Comment by minhnguyent546** [comment]
@itakurah Thank for reviewing. I have created the PR here #3.
